Source/ACE/Factories/GeneratorFactory.cs

```diff
@@ -85,38 +85,65 @@ public static List<WorldObject> CreateWorldObjectsFromGenerator(AceObject genera
                         break;
                 }
 
-                AceObject baseObject = DatabaseManager.World.GetAceObjectByWeenie(generator.ActivationCreateClass);
-                baseObject.Generator = generator.AceObjectId;
-
-                // Determine the ObjectType and call the specific Factory
-                ObjectType ot = (ObjectType)baseObject.Type;
-                switch (ot)
+                // If this generator has linked generators use those for spawning objects
+                if (generator.ActivationCreateClass == 0)
                 {
-                    case ObjectType.Creature:
-                        // TODO: enhance this if we need to spawn NPCs too, else it is just monsters for this tpye
-                        results.Add(MonsterFactory.SpawnMonster(baseObject, pos));
-                        break;
-
-                    case ObjectType.Portal:
-                        // TODO: enable generators to spawn portals, i.e. for Humming Crystal Portal
-                        // results.Add();
-                        break;
+                    // Spawn this generator if it's not the top-level generator
+                    if (generator.Generator != null)
+                    {
+                        results.Add(new Generator(new ObjectGuid(GuidManager.NewItemGuid()), generator));
+                        generator.GeneratorEnteredWorld = true;
+                    }
+
+                    // Get a random generator from the weighted list of linked generators and read it's AceObject from the DB
+                    if (generator.GeneratorLinks.Count == 0)
+                        return null;
+                    uint linkId = GetRandomGeneratorIdFromGeneratorList(random, generator.GeneratorLinks);
+                    var newGen = DatabaseManager.World.GetAceObjectByWeenie(linkId);
```

Comment:
IMO var should only be used when the type is defined as part of the statements rvalue:

eg: var myThing = new ReallyLongClassNameThatsAnnoyingToType();

Not blocking PR over it, but please change.